ntp-proto/src/time_types.rs

```diff
@@ -381,6 +382,47 @@ ntp_duration_scalar_div!(u16);
 ntp_duration_scalar_div!(u32);
 // u64 and usize deliberately excluded as they can result in overflows
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct PollInterval(i8);
+
+impl PollInterval {
+    pub const MIN: Self = Self(4);
+    pub const MAX: Self = Self(17);
+
+    pub fn inc(&mut self) {
+        *self = Self(self.0 + 1).min(Self::MAX);
+    }
+
+    #[must_use]
```

Comment:
So far `inc()` and `dec()` only appear to be used in tests, whereas `plus_one()` is actually used in the library. Will that change? I would probably only have copying `inc()` and `dec()` and ditch `plus_one()`; having both seems a little weird and modifying by `&mut` isn't actually that valuable for a type that's smaller than a word.

Reply:
I pulled this from branch, but turns out inc/dec also were not used there. I've changed the semantics of inc/dec to take/return by-value